src/engine.cairo

```diff
@@ -50,9 +50,18 @@ pub trait EngineTrait {
     fn has_flag(ref self: Engine, flag: ScriptFlags) -> bool;
     // Return the script since last OP_CODESEPARATOR
     fn sub_script(ref self: Engine) -> ByteArray;
+
+    // check the stack size before pushing data
+    fn check_stack_size(ref self: Engine) -> Result<bool, felt252>;
 }
 
 pub impl EngineImpl of EngineTrait {
+    fn check_stack_size(ref self: Engine) -> Result<bool, felt252> {
+        if self.dstack.len() >= 1000 {
```

Comment:
Can you make the 1000 a constant defined at the top of the file maybe, `MAX_STACK_SIZE`.